src/services/api.tsx

```diff
@@ -19,55 +100,182 @@ const getAge = (birthDate: Date) => {
   return age;
 };
 
-/* getCount function returns the number of resources of a type"; */
-export const getCount = async (resource: string, queryParameters: object) => {
-  const response = await client.search({
-    type: resource,
-    query: queryParameters
-  });
-  return response.data.total;
+export const getPatients = async (
+  param?: string,
+  count = PATIENT_REQUESTED
+) => {
+  /*
+   * getPatients return the lists of patients depending on param.
+   * param (optional): if empty, will return the whole list.
+   */
+
+  let response: PatientBundle = (await makeRequest(
+    "Patient",
+    false,
+    param,
+    count
+  )) as PatientBundle;
+
+  return addPatientsToBundle(response);
 };
 
-export const getPatients = async () => {
-  const response = await client.search({
-    type: "Patient",
-    query: { _count: 100 }
-  });
+export const getPatientsPerQuery = async (
+  searchName: String,
+  searchParams: any
+) => {
+  /*
+   * Function getPatientsPerQuery used to get PatientBundle according to query parameters
+   * searchNameParams: object containing name parameters search
+   * searchParams: object[] containing list of other parameters
+   * TODO : for now, this function is limited by the server response to 500 resources. Must find a way to improve it (searching computation done by the server ideally ?)
+   */
+  let bundles: Bundle[] = [];
+  let params: string;
 
-  return response.data.entry.map(
-    ({ resource: { id, identifier, birthDate, name } }: any) => {
-      const patient: Patient = {
-        id: id,
-        age: birthDate && getAge(new Date(birthDate))
-      };
-      if (name) {
-        if (name[0].given) patient.firstName = name[0].given.join(", ");
-        if (name[0].family) patient.lastName = name[0].family;
-      }
-      if (identifier) {
-        patient.identifier = identifier
-          .map((e: any) => {
-            return e.value;
-          })
-          .join(", ");
-      }
+  if (searchName) {
+    params = "";
 
-      return patient;
-    }
+    searchName.split(" ").map((x: string) => {
+      params += "&name=" + x;
+      return params;
+    });
```

Comment:
I think
```
searchName.split(" ").map((x: string) => `${params}&name=`{x}`
```

Reply:
But I don't really get it, what do you do with the list resulting from the map?

Reply:
discussed, replaced by : 
```

    params += searchName
      .split(" ")
      .map((x: string) => `&identifier=${x}`)
      .join();
```